frontend/src/metabase/visualizations/echarts/cartesian/model/axis.ts

```diff
@@ -448,15 +449,35 @@ export function getYAxesModels(
   };
 }
 
+function getXAxisExtent(
```

Comment:
It would be nice to reuse in any way `getDatasetExtents` function we have in `./dataset.ts` to avoid having multiple functions that generally do the same thing. Maybe we could have `getSeriesExtent` which will be used to create the extents map in `getDatasetExtents`